torchgeo/datasets/utils.py

```diff
@@ -46,6 +48,49 @@
 )
 
 
+class DatasetNotFoundError(FileNotFoundError):
+    """Raised when a dataset is requested but doesn't exist.
+
+    .. versionadded:: 0.6
+    """
+
+    def __init__(self, dataset: Dataset[object]) -> None:
+        """Intstantiate a new DatasetNotFoundError instance.
+
+        Args:
+            dataset: The dataset that was requested.
+        """
+        msg = "Dataset not found"
```

Comment:
Lot of possible alternatives here. Could even add the name of the dataset to the error message if that makes things more clear.